app/src/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -230,30 +228,18 @@ public static boolean scaleImage(File originalImage, String finalFilePath, int m
             return false;
         }
 
-        // Create a bitmap out of the image file to see if we need to scale it down
         Bitmap bitmap = BitmapFactory.decodeFile(originalImage.getAbsolutePath());
-        int height = bitmap.getHeight();
-        int width = bitmap.getWidth();
-        int largerDimen = Math.max(height, width);
-        int smallerDimen = Math.min(height, width);
-        if (largerDimen > maxDimen) {
-            // If the larger dimension exceeds our max dimension, scale down accordingly
-            double aspectRatio = ((double) smallerDimen) / largerDimen;
-            largerDimen = maxDimen;
-            smallerDimen = (int) Math.floor(maxDimen * aspectRatio);
-            if (width > height) {
-                bitmap = Bitmap.createScaledBitmap(bitmap, largerDimen, smallerDimen, false);
-            } else {
-                bitmap = Bitmap.createScaledBitmap(bitmap, smallerDimen, largerDimen, false);
-            }
+        Bitmap scaledBitmap = getScaledBitmap(bitmap, maxDimen, false);
+        if (scaledBitmap != null) {
             // Write this scaled bitmap to the final file location
             FileOutputStream out = null;
             try {
                 out = new FileOutputStream(finalFilePath);
-                bitmap.compress(type.getCompressFormat(), 100, out);
-                scaledImage = true;
+                scaledBitmap.compress(type.getCompressFormat(), 100, out);
+                return true;
             } catch (Exception e) {
                 e.printStackTrace();
+                return false;
```

Comment:
java question: will the `finally` block still get executed if there's a return statement in the `catch`?